keep/providers/gcpmonitoring_provider/gcpmonitoring_provider.py

```diff
@@ -49,19 +77,91 @@ class GcpmonitoringProvider(BaseProvider):
 
     PROVIDER_DISPLAY_NAME = "GCP Monitoring"
     FINGERPRINT_FIELDS = ["incident_id"]
+    PROVIDER_SCOPES = [
+        ProviderScope(
+            name="roles/logs.viewer",
+            description="Read access to GCP logging",
+            mandatory=True,
+            alias="Logs Viewer",
+        ),
+    ]
 
     def __init__(
         self, context_manager: ContextManager, provider_id: str, config: ProviderConfig
     ):
         super().__init__(context_manager, provider_id, config)
+        self._service_account_data = json.loads(
+            self.authentication_config.service_account_json
+        )
+        self._client = None
 
     def validate_config(self):
-        """
-        Validates required configuration for Prometheus's provider.
-        """
-        # no config
+        self.authentication_config = GcpmonitoringProviderAuthConfig(
+            **self.config.authentication
+        )
+
+    def dispose(self):
         pass
 
+    def validate_scopes(self) -> dict[str, bool | str]:
+        scopes = {}
+        # try initializing the client to validate the scopes
+        try:
+            self.client
```

Comment:
This will invoke lazy client and check permissions, right?

Reply:
Fixed something there!